python-backend/math_utils.py

```diff
@@ -12,91 +14,150 @@
 logger = logging.getLogger('rm_web_app')
 
 
-def generalized_computed_values(a: sympy.core, b: sympy.core, symbol: Symbol, iterations: int = 500) -> (
-        list[mpmath.mpf], list[mpmath.mpf]):
+def generalized_computed_values(a: Callable, b: Callable, iterations: int = 500) -> tuple[
+    OrderedDict[int, mpmath.mpf], OrderedDict[int, mpmath.mpf], OrderedDict[int, mpmath.mpf]]:
     """
     Compute values at each step iteratively for a polynomial continued fraction
     Parameters
     ----------
     :param iterations: computation depth
     :param b: partial numerator provided by user
     :param a: partial denominator provided by user
-    :param symbol: the variable used by the user in numerator and denominator
 
     Returns
     -------
     list of values computed at each step from n = 1 to the number of iterations provided
     """
     # for complex a and b (note that the article reverses a and b)
     # see https://en.wikipedia.org/wiki/Generalized_continued_fraction#
-    a_n = [mpmath.mpf(a.subs(symbol, n).evalf(PRECISION)) for n in range(0, iterations)]
-    b_n = [mpmath.mpf(b.subs(symbol, n).evalf(PRECISION)) for n in range(0, iterations)]
+    a_n = [a(n) for n in range(0, 2 * iterations)]
```

Comment:
doubling the iteration count facilitates approximation of a limit

Reply:
This comment would be good to add to the code.